src/ol/style/regularshape.js

```diff
@@ -401,11 +431,12 @@ ol.style.RegularShape.prototype.draw_ = function(renderOptions, context, x, y) {
   context.translate(x, y);
 
   context.beginPath();
+  var points = this.points_;
   if (this.radius2_ !== this.radius_) {
-    this.points_ = 2 * this.points_;
+    points = 2 * points;
```

Comment:
This will cause a change of behaviour of the getPoints method:
It will now always return the initially given number of points, regardless of radius2 was given to the constructor or not.